### ISSUE TITLE
Produce a workflow on how to manage the version using the github action.

### ISSUE THREAD

**Issue by darwinsubramaniam** [udwtech/MarketShareCompanion#12]

**Comment by darwinsubramaniam** [comment]
There is one Automated Version Bump action found [Github Action](https://github.com/marketplace/actions/automated-version-bump ).  Using this can enable automated checking based on the commit messages. 

By applying this process the git commit message become very important to indicate if the version is major or minor release. 

### **Workflow**

 Based on the commit messages, increment the version from the latest release.
       
-  If the string "BREAKING CHANGE" or "major" is found anywhere in any of the commit messages or descriptions the major version will be incremented.
-  If a commit message begins with the string "feat" or includes "minor" then the minor version will be increased.
 This works for most common commit metadata for feature additions: "feat: new API" and "feature: new API".
- All other changes will increment the patch version.

- Push the bumped npm version in package.json back into the repo.

- Push a tag for the new version back into the repo.

**Comment by darwinsubramaniam** [comment]
The above Automated Version Bump have shown the disadvantage of monorepo, this is dangerous especially when the changes is focused at one component. The version bump is taking place in overall project. This is undesirable. This brings the split of repository to have multiple plugins as respective repo. The one required work required is also a nest library for extractor plugin. This will ensure how the Task Scheduler will able to call the extractor to activate certain plugins . 
This public API facing between the taskscheduler and the extractor should be finalized before starting the next development . 
The public API facing between the taskscheduler and the AIFactory should also be finalized but this can be push to lower priority as the requirement of collecting the data have to be prioritized first.
